### ISSUE TITLE
java.lang.IllegalArgumentException: ByteBuffer is not a valid flatbuffer model

### ISSUE THREAD

**Issue by samaa** [tensorflow/tensorflow#47051]
Please, in desperate need for help, has been trying to solve for 10 days. the tensorflow lite model I trained is here  [here](uhttps://drive.google.com/file/d/1fYay6FXNlAXi-migvGRtaeqNXB8gKZXk/view?usp=sharingrl). I ran python inference test and it worked. However, no way it is working on Android sample object detection app here https://github.com/tensorflow/examples/tree/master/lite/examples/object_detection/android

by debugging the issue this  long modelHandle = createModelWithBuffer(this.modelByteBuffer, errorHandle); specifically this part in  NativeInterpreterWrapper.class
 NativeInterpreterWrapper(ByteBuffer buffer, Options options) {
        this.inferenceDurationNanoseconds = -1L;
        this.isMemoryAllocated = false;
        this.delegates = new ArrayList();
        this.ownedDelegates = new ArrayList();
        TensorFlowLite.init();
        if (buffer != null && (buffer instanceof MappedByteBuffer || buffer.isDirect() && buffer.order() == ByteOrder.nativeOrder())) {
            this.modelByteBuffer = buffer;
            long errorHandle = createErrorReporter(512);
            long modelHandle = createModelWithBuffer(this.modelByteBuffer, errorHandle);
            this.init(errorHandle, modelHandle, options);
        } else {
            throw new IllegalArgumentException("Model ByteBuffer should be either a MappedByteBuffer of the model file, or a direct ByteBuffer using ByteOrder.nativeOrder() which contains bytes of model content.");
        }
    }

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform MACOS sieera 10.13
- Android studio 4
- I have tried every possible solution and updated the NDK
I used in the gradle

buildscript {
    repositories {
        google()
        jcenter()
        mavenLocal()
    }

aaptOptions {
        noCompress "tflite"
        noCompress "lite"
    }

  implementation 'org.tensorflow:tensorflow-lite-metadata:0.0.0-nightly'
    implementation 'org.tensorflow:tensorflow-lite-select-tf-ops:0.0.0-nightly'
also tried
 implementation 'org.tensorflow:tensorflow-lite-metadata:0.1.2-nightly'  and no difference

please help me I don't know if it is the file it self or the android libraries

**Comment by teijeong** [comment]
Hi, please describe error in detail.

**Comment by samaa** [comment]
Hello, I have solved it. I tried the following 
comment the 
//apply from:'download_model.gradle'
in app gridle. Don't try to download the model from a link.
Also, I removed that part from TFLiteObjectDetectionAPIModel

try (BufferedReader br =
    new BufferedReader(
        new InputStreamReader(
            metadata.getAssociatedFile(labelFilename), Charset.defaultCharset()))) {
  String line;
  while ((line = br.readLine()) != null) {
    Log.w(TAG, line);
    d.labels.add(line);
  }
}

instead I added labels to the "d" directly and those labels are the classes of the custom trained model
d.labels.add("driver");
d.labels.add("passenger");
